### DISCUSSION TITLE
ISR vs SWR

### DISCUSSION THREAD

**Discussion by dbzx10299** [nuxt/nuxt#28500] [Questions]
Can some **please** explain the difference to me from swr and isr with nuxt. I am having a hard time understanding what the difference is and I think the docs might be wrong or nextjs might have a different implementation.

It's clear to me what swr is and I understand it perfectly. When I deploy a site to netlify with swr I can view the cache headers and see a header `Cache-Status: "Netlify Edge"; hit` this means the response is indeed in the edge.

The docs state that swr and isr are basically the same only isr stores the response on the edge node,  but looking at my http headers I can clearly see that the response is already stored in the edge node using swr so I don't see the difference.

Also I looked at this [vercel isr demo](https://github.com/danielroe/nuxt-vercel-isr) example from @danielroe and it seems to be working like swr to me.  Is this because nextjs and nuxtjs have different versions of what is considered isr?

**Comment by arashsheyda** [comment]
@dbzx10299 I found [this article](https://blog.risingstack.com/nuxt-3-rendering-modes/)

**Comment by oneminch** [comment]
According to the article shared by @arashsheyda, ISR in Nuxt is different from the one in Next.js. In Nuxt ISR, HTML is generated _on demand_. In Next.js ISR, HTML is typically generated during build time by default. 

As I understand it, the main difference between SWR and ISR is that SWR caches responses on the _server_ while ISR does it on the _CDN_. However, with hosting providers (like Netlify), the line between SWR and ISR is blurred (since they cache SWR responses at the edge and the edge is basically a bunch _servers on a CDN_).

**Reply by dbzx10299** [reply]
From what I understood with reading about ISR using nextjs, ISR was intended so build times weren‘t unnecessarily long. That means the pages were generated on demand then wouldn‘t be regenerated until either content changed or until it became stale.

**Comment by anatoliy-larin** [comment]
Hi there,

good article about rendering modes https://blog.risingstack.com/nuxt-3-rendering-modes/
